src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -315,7 +317,14 @@ public void logReEvaluate(User user, ProgrammingExercise exercise, Course course
      * @return testCases, but the ones based on the described visibility criterion removed.
      */
     private Set<ProgrammingExerciseTestCase> filterTestCasesForCurrentDate(ProgrammingExercise exercise, Set<ProgrammingExerciseTestCase> testCases) {
-        boolean isBeforeDueDate = exercise.isBeforeDueDate();
+        boolean isBeforeDueDate;
+        if (exercise.isExamExercise()) {
```

Comment:
I suggest to move the whole new code to the `examDateService` so that it can be used in more places